Oh, multiple jobs failed when running `https://apt.llvm.org/llvm.sh`:

```
+ add-apt-repository -y 'deb https://apt.llvm.org/resolute/  llvm-toolchain-resolute-19 main'
Get:1 file:/etc/apt/apt-mirrors.txt Mirrorlist [144 B]
Hit:6 https://packages.microsoft.com/repos/azure-cli jammy InRelease
Hit:7 https://packages.microsoft.com/ubuntu/26.04/prod resolute InRelease
Hit:8 https://dl.google.com/linux/chrome-stable/deb stable InRelease
Hit:2 http://azure.archive.ubuntu.com/ubuntu resolute InRelease
Hit:3 http://azure.archive.ubuntu.com/ubuntu resolute-updates InRelease
Hit:4 http://azure.archive.ubuntu.com/ubuntu resolute-backports InRelease
Hit:5 http://azure.archive.ubuntu.com/ubuntu resolute-security InRelease
Ign:9 https://apt.llvm.org/resolute llvm-toolchain-resolute-19 InRelease
Err:10 https://apt.llvm.org/resolute llvm-toolchain-resolute-19 Release
  404  Not Found [IP: 146.75.106.49 443]
Reading package lists...
E: The repository 'https://apt.llvm.org/resolute llvm-toolchain-resolute-19 Release' does not have a Release file.
Repository: 'deb https://apt.llvm.org/resolute/ llvm-toolchain-resolute-19 main'
Description:
Archive for codename: llvm-toolchain-resolute-19 components: main
More info: https://apt.llvm.org/resolute/
Adding repository.
Adding deb entry to /etc/apt/sources.list.d/archive_uri-https_apt_llvm_org_resolute_-resolute.list
Adding disabled deb-src entry to /etc/apt/sources.list.d/archive_uri-https_apt_llvm_org_resolute_-resolute.list

+ apt-get update
Get:1 file:/etc/apt/apt-mirrors.txt Mirrorlist [144 B]
Hit:2 http://azure.archive.ubuntu.com/ubuntu resolute InRelease
Hit:3 http://azure.archive.ubuntu.com/ubuntu resolute-updates InRelease
Hit:6 https://packages.microsoft.com/repos/azure-cli jammy InRelease
Hit:4 http://azure.archive.ubuntu.com/ubuntu resolute-backports InRelease
Hit:5 http://azure.archive.ubuntu.com/ubuntu resolute-security InRelease
Hit:7 https://packages.microsoft.com/ubuntu/26.04/prod resolute InRelease
Ign:8 https://apt.llvm.org/resolute llvm-toolchain-resolute-19 InRelease
Err:9 https://apt.llvm.org/resolute llvm-toolchain-resolute-19 Release
  404  Not Found [IP: 146.75.106.49 443]
Hit:10 https://dl.google.com/linux/chrome-stable/deb stable InRelease
Reading package lists...
E: The repository 'https://apt.llvm.org/resolute llvm-toolchain-resolute-19 Release' does not have a Release file.
Error: Process completed with exit code 100.
```

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -272,9 +272,11 @@ jobs:
         free-threading:
         - false
         - true
+        # For BOLT jobs, https://apt.llvm.org/llvm.sh doesn't support LLVM 19
+        # on Ubuntu 26.04, so stick to Ubuntu 24.04 until LLVM is upgraded.
         os:
-        - ubuntu-26.04
-        - ubuntu-26.04-arm
+        - ubuntu-24.04
+        - ubuntu-24.04-arm
         exclude:
         # Do not test BOLT with free-threading, to conserve resources
         - bolt: true

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -282,7 +282,7 @@ jobs:
         - bolt: true
           free-threading: true
         # BOLT currently crashes during instrumentation on aarch64
-        - os: ubuntu-26.04-arm
+        - os: ubuntu-24.04-arm
           bolt: true
     uses: ./.github/workflows/reusable-ubuntu.yml
     with:

--- a/.github/workflows/jit.yml
+++ b/.github/workflows/jit.yml
@@ -161,7 +161,9 @@ jobs:
   linux-extras:
     name: ${{ matrix.name }}
 
-    runs-on: ubuntu-26.04
+    # https://apt.llvm.org/llvm.sh doesn't support LLVM 19 on Ubuntu 26.04,
+    # so stick to Ubuntu 24.04 until LLVM is upgraded.
+    runs-on: ubuntu-24.04
     timeout-minutes: 60
     strategy:
       fail-fast: false
@@ -184,7 +186,8 @@ jobs:
           python-version: '3.11'
       - name: Build
         run: |
-          # On ubuntu-26.04 image, clang is clang-21 by default
+          sudo bash -c "$(wget -O - https://apt.llvm.org/llvm.sh)" ./llvm.sh ${{ env.LLVM_VERSION }}
+          export PATH="$(llvm-config-${{ env.LLVM_VERSION }} --bindir):$PATH"
           if [ "${{ matrix.use_clang }}" = "true" ]; then
             export CC=clang-${{ env.LLVM_VERSION }}
           fi

--- a/.github/workflows/jit.yml
+++ b/.github/workflows/jit.yml
@@ -138,10 +138,12 @@ jobs:
           - true
           - false
         include:
+          # https://apt.llvm.org/llvm.sh doesn't support LLVM 19
+          # on Ubuntu 26.04, so stick to Ubuntu 24.04 until LLVM is upgraded.
           - target: x86_64-unknown-linux-gnu/gcc
-            runner: ubuntu-26.04
+            runner: ubuntu-24.04
           - target: aarch64-unknown-linux-gnu/gcc
-            runner: ubuntu-26.04-arm
+            runner: ubuntu-24.04-arm
     steps:
       - uses: actions/checkout@de0fac2e4500dabe0009e67214ff5f5447ce83dd # v6.0.2
         with:
@@ -151,7 +153,8 @@ jobs:
           python-version: '3.11'
       - name: Build
         run: |
-          # On ubuntu-26.04 image, clang is clang-21 by default
+          sudo bash -c "$(wget -O - https://apt.llvm.org/llvm.sh)" ./llvm.sh ${{ env.LLVM_VERSION }}
+          export PATH="$(llvm-config-${{ env.LLVM_VERSION }} --bindir):$PATH"
           ./configure --enable-experimental-jit ${{ matrix.debug && '--with-pydebug' || '' }}
           make all --jobs 4
       - name: Test

--- a/.github/workflows/posix-deps-apt.sh
+++ b/.github/workflows/posix-deps-apt.sh
@@ -13,7 +13,6 @@ apt-get -yq --no-install-recommends install \
     libgdbm-compat-dev \
     libgdbm-dev \
     liblzma-dev \
-    libmpdec-dev \
     libncurses5-dev \
     libreadline6-dev \
     libsqlite3-dev \

--- a/.github/workflows/reusable-san.yml
+++ b/.github/workflows/reusable-san.yml
@@ -27,7 +27,9 @@ jobs:
         && ' (free-threading)'
         || ''
       }}
-    runs-on: ubuntu-26.04
+    # https://apt.llvm.org/llvm.sh doesn't support LLVM 20 on Ubuntu 26.04,
+    # so stick to Ubuntu 24.04 until LLVM is upgraded.
+    runs-on: ubuntu-24.04
     timeout-minutes: 60
     steps:
     - uses: actions/checkout@de0fac2e4500dabe0009e67214ff5f5447ce83dd # v6.0.2

--- a/.github/workflows/reusable-ubuntu.yml
+++ b/.github/workflows/reusable-ubuntu.yml
@@ -43,6 +43,8 @@ jobs:
       run: sudo ./.github/workflows/posix-deps-apt.sh
     - name: Install Clang and BOLT
       if: ${{ fromJSON(inputs.bolt-optimizations) }}
+      # For BOLT jobs, https://apt.llvm.org/llvm.sh doesn't support LLVM 19
+      # on Ubuntu 26.04, so stick to Ubuntu 24.04 until LLVM is upgraded.
       run: |
         sudo bash -c "$(wget -O - https://apt.llvm.org/llvm.sh)" ./llvm.sh 19
         sudo apt-get install --no-install-recommends bolt-19

